### PR TITLE
feat: track total hits in ES7 if configured

### PR DESCRIPTION
**Pull request by jywadhwani**
The default behavior of ES currently is to _not_ track the total number of hits if more than 10,000 (https://www.elastic.co/guide/en/elasticsearch/reference/7.13/search-your-data.html#track-total-hits) This change does the following
1. Changes the default behavior to return the exact hits
2. Exposes methods to set the lower bound of hits  

## Checklist

- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [X] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java [hunk 1]
```diff
@@ -48,12 +48,35 @@
 public class ESBrowseDAO extends BaseBrowseDAO {
   private final RestHighLevelClient _client;
   private final BaseBrowseConfig _config;
+  private static final Integer ES_LOWER_BOUND_HITS = 10000;
+  private static final Integer DEFAULT_LOWER_BOUND_HITS = 2147483647;
```
Comment (keremsahin1) [inline dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java:52]: You mean Integer.MAX_VALUE here?

File: dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java [hunk 2]
```diff
@@ -48,12 +48,35 @@
...
     this._client = esClient;
     this._config = config;
   }
 
+  /**
+   * Set "track_total_hits" query parameter to false if you do not need accurate results. It is a good trade off to
+   * speed up searches if you don’t need the accurate number of hits after a certain threshold. If set to false, it will
+   * use the default lower bound set by Elasticsearch.
+   */
+  public void setTrackTotalHits(boolean trackTotalHits) {
```
Comment (keremsahin1) [inline dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java:65]: Maybe you can get rid of this method and keep the other one. Any user wants to limit max hits could set it accordingly.

File: dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java (hunk 1 above)
Comment (jywadhwani) [reply dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java:52]: This is what ES had set it automatically when using `"track_total_hits": true` in search query. Your suggestion can be incorporated.

File: dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java (hunk 2 above)
Comment (jywadhwani) [reply dao-impl/elasticsearch-dao-7/src/main/java/com/linkedin/metadata/dao/browse/ESBrowseDAO.java:65]: Sure

**Review by keremsahin1 — APPROVED** [review]
LGTM. Thanks for the quick fix!

**Pull request merged**